bigbluebutton-html5/imports/ui/components/whiteboard/annotations/pencil/component.jsx

```diff
@@ -89,13 +89,13 @@ export default class PencilDrawComponent extends Component {
     return version !== nextProps.version;
   }
 
-  componentWillUpdate(nextProps) {
-    const { annotation: nextAnnotation, slideWidth, slideHeight } = nextProps;
-    const { points: nextPoints } = nextAnnotation;
-    const { annotation } = this.props;
+  componentDidUpdate(prevProps) {
+    const { annotation: prevAnnotation } = prevProps;
+    const { points: prevPoints } = prevAnnotation;
+    const { annotation, slideWidth, slideHeight } = this.props;
     const { points } = annotation;
-    if (nextPoints.length !== points.length) {
-      this.path = this.getCoordinates(nextAnnotation, slideWidth, slideHeight);
+    if (prevPoints.length !== points.length) {
+      this.path = this.getCoordinates(annotation, slideWidth, slideHeight);
```

Comment:
This was being calculated before rendering. Are you sure it will work the same as before if now we only get the coordinates after the component renders?

Reply:
I tested this locally and am not spotting issues. Do you know of a specific case where this might get broken? 
@Tainan404 could you please comment too?

Reply:
All I could see was that we got something that previously we calculated before rendering and postponed to calculate after rendering.
So, I would expect the component to not render newer paths in some cases.

Reply:
Unless I'm missing something here.